requests_cache/backends/base.py

```diff
@@ -113,12 +113,18 @@ def clear(self):
         self.responses.clear()
         self.keys_map.clear()
 
-    def remove_old_entries(self, created_before):
-        """Deletes entries from cache with creation time older than ``created_before``"""
+    def remove_old_entries(self, expires_before):
+        """Deletes entries from cache with expiration time older than ``expires_before``"""
         keys_to_delete = set()
-        for key, (response, created_at) in self.responses.items():
-            if created_at < created_before:
-                keys_to_delete.add(key)
+        for key, (response, _) in self.responses.items():
+            if response.expiration_date is None:
```

Comment:
Good point, I will look into that!

Reply:
Moved the first check up, under the assumption that expiration_date is always timezone-aware. 
Maybe we should assume that a user might set a non-timezone aware expiration_date manually? (or, actually, as `expire_after`... maybe the `_determine_expiration_datetime` should handle that, or `expiration_date` should become a property with a setter to ensure that it is timezone aware?

Reply:
I'd say the fewer places where we have to worry about timezones, the better. Currently it looks like the only place to pass in a user-provided `datetime` is here, with `expires_before`. `response.expiration_date` should only be created/modified internally.

If you can think of a case where you would need expire something by setting `response.expiration_date` manually and couldn't do it via `request()` or `remove_expired_responses()`, then we could add that. Otherwise I'd prefer to just treat it as an internal value that's always UTC.